base/src/ArchiveSpaceManager.cpp

```diff
@@ -0,0 +1,180 @@
+#include <stdafx.h>
+#include<cstdlib>
+#include <boost/filesystem.hpp>
+#include "Module.h"
+#include "ArchiveSpaceManager.h"
+
+class ArchiveSpaceManager::Detail
+{
+public:
+    Detail(ArchiveSpaceManagerProps& _props) : mProps(_props)
+    {
+    }
+
+    ~Detail()
+    {
+    }
+
+    void setProps(ArchiveSpaceManagerProps _props)
+    {
+        mProps = _props;
+    }
+  
+    uint32_t estimateDirectorySize(boost::filesystem::path _dir)
+    {
+        uint32_t dirSize = 0;
+        int sample = 0;
+        int inCount = 0;
+        int countFreq = 0;
+        uint32_t tempSize = 0;
+       
+        for (const auto& entry : boost::filesystem::recursive_directory_iterator(_dir))
+        {
+            if ((boost::filesystem::is_regular_file(entry)))
+            {
+                if (!(countFreq % mProps.samplingFreq))
+                {
+                    sample = (rand() % mProps.samplingFreq) + 1;
+                    inCount = 0;
+                }
+                if (inCount == sample)
+                {
+                    tempSize = boost::filesystem::file_size(entry);
+                    dirSize += tempSize * mProps.samplingFreq;
+                }
+                inCount++;
+                countFreq++;
+            }
+        }
+        if (inCount < mProps.samplingFreq)
+        {
+            dirSize = dirSize + (tempSize * inCount);
+        }
+        return dirSize;
+    }
+
+    boost::filesystem::path getOldestDirectory(boost::filesystem::path _cam)
+    {
+        for (const auto& camFolder : boost::filesystem::directory_iterator(_cam))
+        {
+            for (const auto& folder : boost::filesystem::recursive_directory_iterator(camFolder))
+            {
+                if (boost::filesystem::is_regular_file(folder))
+                {
+                    boost::filesystem::path p = folder.path().parent_path();
+                    return p;
+                }
+            }
+        }
+    };
+
+    void manageDirectory()
+    {
+        auto comparator = [](std::pair<boost::filesystem::path, uint32_t>& a, std::pair<boost::filesystem::path, uint32_t>& b) {return a.second < b.second; };
+        while (archiveSize > mProps.lowerWaterMark)
+        {
+            for (const auto& camFolder : boost::filesystem::directory_iterator(mProps.pathToWatch))
+            {
+                boost::filesystem::path oldHrDir = getOldestDirectory(camFolder);
+                foldVector.push_back({oldHrDir,boost::filesystem::last_write_time(oldHrDir) });
+            }
+            sort(foldVector.begin(), foldVector.end(), comparator); //Sorting the vector
+
+            uint32_t tempSize = 0;
+            boost::filesystem::path delDir = foldVector[0].first;
+            BOOST_LOG_TRIVIAL(info) <<"Deleting file : "<< delDir.string();
+            tempSize = estimateDirectorySize(delDir);
+            archiveSize = archiveSize - tempSize;
+            boost::filesystem::remove_all(delDir);
```

Comment:
we should handle any delete exception here.

Reply:
Try() Catch(..) is added in the delete logic.